pkg/services/govmomi/vcenter/clone.go

```diff
@@ -397,3 +416,50 @@ func getNetworkSpecs(ctx *context.VMContext, devices object.VirtualDeviceList) (
 
 	return deviceSpecs, nil
 }
+
+func createPCIPassThroughDevice(deviceKey int32, backingInfo types.BaseVirtualDeviceBackingInfo) types.BaseVirtualDevice {
+	device := &types.VirtualPCIPassthrough{
+		VirtualDevice: types.VirtualDevice{
+			Key:     deviceKey,
+			Backing: backingInfo,
+		},
+	}
+	return device
+}
+
+func getGpuSpecs(
+	ctx *context.VMContext) ([]types.BaseVirtualDeviceConfigSpec, error) {
+	deviceSpecs := []types.BaseVirtualDeviceConfigSpec{}
+	deviceKey := int32(-200)
+
+	expectedPciDevices := ctx.VSphereVM.Spec.VirtualMachineCloneSpec.PciDevices
+	if len(expectedPciDevices) == 0 {
+		return nil, errors.Errorf("Invalid pci device count count: %d", len(expectedPciDevices))
+	}
+
+	for _, pciDevice := range expectedPciDevices {
+		if pciDevice.VendorID == nil {
+			return nil, errors.Errorf("unable to get vendorID")
+		}
+		if pciDevice.DeviceID == nil {
+			return nil, errors.Errorf("unable to get deviceID")
+		}
```

Comment:
This can be removed, since they will not be nil after the kubebuilder validation mentioned above.